### PR TITLE
Producer tests

### LINKED ISSUE #365: Cover producer with tests

### PR DESCRIPTION
**Pull request by JoseLSegura**
# Description

A little refactor to the producer package and adding unit tests

Fixes #365 

## Type of change

- Refactor (refactoring code, removing useless files)
- Unit tests (no changes in the code)

## Testing steps
CI

### CODE REVIEW

File: broker/configuration.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ import (
 type Configuration struct {
 	Address      string     `mapstructure:"address" toml:"address"`
 	Topic        string     `mapstructure:"topic" toml:"topic"`
+	PublishTopic string     `mapstructure:"publish_topic" toml:"publish_topic"`
```
Comment (Sergey1011010) [inline broker/configuration.go]: Why not to use `Topic`?

File: producer/producer.go [hunk 2]
```diff
@@ -28,31 +28,55 @@ import (
...
+	return &KafkaProducer{
+		Configuration: brokerCfg,
+		Producer:      producer,
+	}, nil
+}
+
+// ProduceMessage produces message to selected topic. That function returns
+// partition ID and offset of new message or an error value in case of any
+// problem on broker side.
+func (producer *KafkaProducer) ProduceMessage(message string) (partition int32, offset int64, errout error) {
```
Comment (Sergey1011010) [inline producer/producer.go]: why do you use named return value here? It doesn't seem to be necessary.

File: producer/producer_test.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+	mockProducer := mocks.NewSyncProducer(t, nil)
+	mockProducer.ExpectSendMessageAndSucceed()
+	producer := producer.KafkaProducer{
+		Configuration: brokerCfg,
+		Producer:      mockProducer,
+	}
+
+	_, _, err := producer.ProduceMessage("Hello world")
+	assert.Equal(t, err, nil)
```
Comment (Sergey1011010) [inline producer/producer_test.go]: In most of cases like this we use helpers.FailOnError(t, err) so it's better to follow one standard :)

File: broker/configuration.go (hunk 1 above)
Comment (JoseLSegura) [reply broker/configuration.go]: When we need to use the `producer` package probably we want to use a different topic for consuming and producing messages

File: producer/producer.go (hunk 2 above)
Comment (JoseLSegura) [reply producer/producer.go]: Agree

File: producer/producer_test.go (hunk 3 above)
Comment (JoseLSegura) [reply producer/producer_test.go]: Ok! Thank you!

**Pull request merged**